src/tls13.c

```diff
@@ -3409,18 +3578,37 @@ int DoTls13ServerHello(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
     args->idx += OPAQUE16_LEN;
 
 #ifndef WOLFSSL_NO_TLS12
-    if (args->pv.major == ssl->version.major &&
-        args->pv.minor < TLSv1_2_MINOR &&
-        ssl->options.downgrade)
     {
-        /* Force client hello version 1.2 to work for static RSA. */
-        ssl->chVersion.minor = TLSv1_2_MINOR;
-        ssl->version.minor = TLSv1_2_MINOR;
-        return DoServerHello(ssl, input, inOutIdx, helloSz);
+        byte wantDowngrade;
+
+        wantDowngrade = args->pv.major == ssl->version.major &&
+            args->pv.minor < TLSv1_2_MINOR;
+
+#ifdef WOLFSSL_DTLS13
+        if (ssl->options.dtls)
+            wantDowngrade = args->pv.major == ssl->version.major &&
+                args->pv.minor > DTLSv1_2_MINOR;
+#endif /* WOLFSSL_DTLS13 */
+
+        if (wantDowngrade && ssl->options.downgrade) {
+            /* Force client hello version 1.2 to work for static RSA. */
+            ssl->chVersion.minor = TLSv1_2_MINOR;
+            ssl->version.minor = TLSv1_2_MINOR;
+
+#ifdef WOLFSSL_DTLS13
+            if (ssl->options.dtls) {
+                ssl->chVersion.minor = DTLSv1_2_MINOR;
+                ssl->version.minor = DTLSv1_2_MINOR;
+            }
+#endif /* WOLFSSL_DTLS13 */
+
+            return DoServerHello(ssl, input, inOutIdx, helloSz);
+        }
     }
 #endif
+
     if (args->pv.major != ssl->version.major ||
-        args->pv.minor != TLSv1_2_MINOR) {
+        args->pv.minor != tls12minor) {
```

Comment:
This is happening with async. `byte tls12minor;` should be moved to `struct Dsh13Args`
```
./configure --enable-all --enable-asynccrypt && make

    build...src/tls13.c: In function ‘DoTls13ServerHello’:
src/tls13.c:3611:27: error: ‘tls12minor’ undeclared (first use in this function)
 3611 |         args->pv.minor != tls12minor) {
      |                           ^~~~~~~~~~
```